worlds/pokemon_emerald/__init__.py

```diff
@@ -0,0 +1,841 @@
+
+        if self.multiworld.npc_gifts[self.player].value == Toggle.option_true:
+            set_npc_gift_rules(self.multiworld, self.player)
+
+        if self.multiworld.enable_ferry[self.player].value == Toggle.option_true:
+            set_enable_ferry_rules(self.multiworld, self.player)
+
+        if self.multiworld.require_itemfinder[self.player].value == Toggle.option_true:
+            add_hidden_item_itemfinder_rules(self.multiworld, self.player)
+
+        if self.multiworld.require_flash[self.player].value == Toggle.option_true:
+            add_flash_rules(self.multiworld, self.player)
+
+    def generate_basic(self) -> None:
+        victory_event_name = "EVENT_DEFEAT_CHAMPION"
+        if self.multiworld.goal[self.player] == Goal.option_steven:
+            victory_event_name = "EVENT_DEFEAT_STEVEN"
+        if self.multiworld.goal[self.player] == Goal.option_norman:
+            victory_event_name = "EVENT_DEFEAT_NORMAN"
+
+        self.multiworld.completion_condition[self.player] = lambda state: state.has(victory_event_name, self.player)
+
+        locations: List[PokemonEmeraldLocation] = self.multiworld.get_locations(self.player)
+
+        # Set our free fly location
+        # If not enabled, set it to Littleroot Town by default
+        fly_location_name = "EVENT_VISITED_LITTLEROOT_TOWN"
+        if self.multiworld.free_fly_location[self.player].value == Toggle.option_true:
+            fly_location_name = self.random.choice([
+                "EVENT_VISITED_SLATEPORT_CITY",
+                "EVENT_VISITED_MAUVILLE_CITY",
+                "EVENT_VISITED_VERDANTURF_TOWN",
+                "EVENT_VISITED_FALLARBOR_TOWN",
+                "EVENT_VISITED_LAVARIDGE_TOWN",
+                "EVENT_VISITED_FORTREE_CITY",
+                "EVENT_VISITED_LILYCOVE_CITY",
+                "EVENT_VISITED_MOSSDEEP_CITY",
+                "EVENT_VISITED_SOOTOPOLIS_CITY",
+                "EVENT_VISITED_EVER_GRANDE_CITY"
+            ])
+
+        self.free_fly_location_id = location_visited_event_to_id_map[fly_location_name]
+
+        free_fly_location_location = self.multiworld.get_location("FREE_FLY_LOCATION", self.player)
+        free_fly_location_location.locked = False
+        free_fly_location_location.item = None
+        free_fly_location_location.place_locked_item(self.create_event(fly_location_name))
+
+        # Key items which are considered in access rules but not randomized are converted to events and placed
+        # in their vanilla locations so that the player can have them in their inventory for logic.
+        def convert_unrandomized_items_to_events(tag: str) -> None:
+            for location in locations:
+                if location.tags is not None and tag in location.tags:
+                    location.place_locked_item(self.create_event(self.item_id_to_name[location.default_item_code]))
+                    location.address = None
+                    location.is_event = True
```

Comment:
`place_locked_item` already does this

Reply:
This was a custom attribute, not `Location.event`. It was ultimately duplicate information for basically no benefit, so it was removed in 04a942a.